Backpack/src/main/java/net/skyscanner/backpack/calendar2/BpkCalendar.kt

```diff
@@ -0,0 +1,83 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.calendar2
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.appcompat.widget.LinearLayoutCompat
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.plus
+import net.skyscanner.backpack.calendar2.adapter.CalendarAdapter
+import net.skyscanner.backpack.calendar2.adapter.CalendarSpanSizeLookup
+import net.skyscanner.backpack.calendar2.data.CalendarStateMachine
+import net.skyscanner.backpack.calendar2.view.CalendarHeaderView
+import net.skyscanner.backpack.util.ResourcesUtil
+import org.threeten.bp.LocalDate
+import org.threeten.bp.Period
+
+class BpkCalendar private constructor(
```

Comment:
is there a callback for the consumer for when the selection updates? also looks like there as a scroll listener as well in the old calendar